common/src/field_conversion.rs

```diff
@@ -48,6 +49,36 @@ pub fn ff_to_ruints<FF: GenericPrimeField<Repr = [u8; 32]>>(ff: Vec<FF>) -> Vec<
     ff.into_iter().map(|f| ff_to_ruint(f)).collect()
 }
 
+pub fn ark_to_spartan_unsafe<AF: ArkPrimeField, FF: GenericPrimeField<Repr = [u8; 32]>>(ark: AF) -> FF {
+    assert_eq!(std::mem::size_of::<AF>(), 32);
+    let ff: FF;
+    unsafe {
+        let inner = access_ark_private(&ark);
+        ff = std::mem::transmute_copy(&inner);
```

Comment:
results in the following compiler error:

```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> common/src/field_conversion.rs:57:14
   |
57 |         ff = std::mem::transmute(&inner);
   |              ^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `&[u8; 32]` (64 bits)
   = note: target type: `FF` (this type does not have a fixed size)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> common/src/field_conversion.rs:67:15
   |
67 |         ark = std::mem::transmute(&inner);
   |               ^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `&[u8; 32]` (64 bits)
   = note: target type: `AF` (this type does not have a fixed size)

For more information about this error, try `rustc --explain E0512`.
warning: `common` (lib) generated 1 warning
error: could not compile `common` (lib) due to 2 previous errors; 1 warning emitted
```

given we know the size is 256-bits (panics if not), I think it would be fine, but seems difficult to communicate that to the compiler (rather than runtime)

Reply:
hmm ok, in that case let's keep it as `transmute_copy` but can we add an assertion that `assert_eq!(std::mem::size_of::<FF>(), 32)`?